Update the similarity algorithm details (#9675)
opensearch-project/documentation-website: _search-plugins/keyword-search.md

Algorithm | Description
`BM25` | The default OpenSearch [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) similarity algorithm. 

> `LegacyBM25` | (Deprecated) The older [LegacyBM25Similarity](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/lucene/similarity/LegacyBM25Similarity.java) implementation. Kept for backward compatibility.

Reviewer: ```suggestion
`LegacyBM25` (Deprecated) | The older [LegacyBM25Similarity](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/lucene/similarity/LegacyBM25Similarity.java) implementation. Kept for backward compatibility.
```

`boolean` | Assigns terms a score equal to their boost value. Use `boolean` similarity when you want the document scores to be based on the binary value of whether the terms match.

### Note

> OpenSearch 3.0 and later defaults to Lucene’s `BM25Similarity`. If required, you can opt in to `LegacyBM25` explicitly.

Reviewer 2: I would highlight the difference.

Lucene recognized that scores were being multiplied by a constant factor and dropped it.

In OpenSearch 3.0, by default, all scores will be lower by a factor of 2.2 (if I remember correctly). It doesn't affect the order of results, unless you do something that depends on the specific values.

Examples include the `min_score` query that discards everything below a given score. It may also have implications for hybrid scoring (though I believe that's usually relative to the max score, so the relative scaling should be consistent).

Author: Updated @msfroh. Thanks

> OpenSearch 3.0 and later defaults to Lucene’s `BM25Similarity`. If required, you can opt in to `LegacyBM25` explicitly.

Reviewer: ```suggestion
OpenSearch 3.0 and later defaults to Lucene's `BM25Similarity`. To use the old version, specify `LegacyBM25` explicitly.
```

Algorithm | Description
`BM25` | The default OpenSearch [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) similarity algorithm. 
`LegacyBM25` (Deprecated) | The older [LegacyBM25Similarity](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/lucene/similarity/LegacyBM25Similarity.java) implementation. Kept for backward compatibility.
`boolean` | Assigns terms a score equal to their boost value. Use `boolean` similarity when you want the document scores to be based on the binary value of whether the terms match.

### Important changes to BM25 scoring in OpenSearch 3.0

> In OpenSearch 3.0, the default similarity algorithm switched from `LegacyBM25Similarity` to Lucene’s native `BM25Similarity`.

Reviewer 3: ```suggestion
In OpenSearch 3.0, the default similarity algorithm changed from `LegacyBM25Similarity` to Lucene's native `BM25Similarity`.
```

This change improves alignment with Lucene standards and simplifies scoring behavior, but it introduces an important difference:

- In `LegacyBM25Similarity`, scores included an extra constant factor of `k₁ + 1` in the numerator of the `BM25` formula.

- In `BM25Similarity`, this constant was removed for cleaner normalization (see [BM25](https://en.wikipedia.org/wiki/Okapi_BM25) and the corresponding [Lucene GitHub issue](https://github.com/apache/lucene/issues/9609)).

- Scores produced by `BM25Similarity` are lower than those produced by `LegacyBM25Similarity`, typically by a factor of about `2.2`.

> - Ranking is unaffected, because the constant factor does not change the relative order of documents.

Reviewer 3: ```suggestion
- Ranking is unaffected because the constant factor does not change the relative order of documents.
```

> - To retain the old scoring behavior, explicitly configure your field or index to use `LegacyBM25` (see Configuring `LegacyBM25Similarity` [section](https://opensearch.org/docs/latest/search-plugins/keyword-search/#configuring-legacy-bm25-similarity)).

Author: Is there a way I can point the heading slug? I have tried something like `#configuring-legacy-bm25-similarity` but dint work

Reviewer: You have to use relative links. Updated.

Author: Thanks

`b` | Float | Determines the degree to which document length normalizes TF values. The default value is `0.75`.
`discount_overlaps` | Boolean | Determines whether overlap tokens (tokens with zero position increment) are ignored when computing the norm. Default is `true` (overlap tokens do not count when computing the norm). 

> ## Configuring LegacyBM25 similarity 

Reviewer: ```suggestion
## Configuring legacy BM25 similarity 
```

> If you want to retain the older similarity behavior:

Reviewer: ```suggestion
If you want to retain the older similarity behavior, specify `LegacyBM25` as the similarity `type`:
```
